internals/cli/app.go

```diff
@@ -265,7 +272,13 @@ func (c *CommandClause) registerEnvVarParsing() {
 // adding an environment variable default configurable by APP_COMMAND_FLAG_NAME.
 // The help text is suffixed with the default value of the flag.
 func (c *CommandClause) Flag(name string) *Flag {
+	if _, ok := c.flags[name]; ok {
```

Comment:
If we assume persistent flags are only set on the root command, I think we can hugely simplify this by just changing the template to:
```
{{flagUsages .App.Root false | trimTrailingWhitespaces}}
```

Then we can even get rid of the `isGlobal` argument. 

Would that work?

Reply:
It did work indeed. Nice tip.
Adjusted properly.

Reply:
Can we now revert d2d6c8d4b12d470ffd6786805177a71e3610218f?

Reply:
Not all the code. Some of it is still important. The commit contains adjustments to the `Hidden()` function, as well as the fix for `mkdir` (allow multiple directories to be provided).
I will make a commit in which the code from the `Flag` function is simplified.
The only part that I left from the code is the following, since I consider it eliminates the case of a flag being repeatedly registered to a certain environment variable:
```
if _, ok := c.flags[name]; ok {
	return c.flags[name]
}
```

Reply:
>  The only part that I left from the code is the following, since I consider it eliminates the case of a flag being repeatedly registered to a certain environment variable:

This was actually the piece of code that I was most concerned about. We're now applying a patch that is only needed (though I think it's not strictly needed) for building the help text to all  cases where `Flag()` is called. `Flag()` is now more complex, only because the help text needs the name of an environment variable. In some way, we're adding the `map[string]*Flag` (and all complexcity that comes with this), just to expose these lines of code:

```
fullCmd := strings.Replace(c.fullCommand(), " ", c.App.separator, -1)
prefix := formatName(fullCmd, "", c.App.separator, c.App.delimiters...)
envVar := formatName(name, prefix, c.App.separator, c.App.delimiters...)
```

Can't we just expose this function and call it in `Flag()` and `flagUsages()`?


```golang
func (c *CommandClause) FlagEnvName(name string) string{
	fullCmd := strings.Replace(c.fullCommand(), " ", c.App.separator, -1)
	prefix := formatName(fullCmd, "", c.App.separator, c.App.delimiters...)
	return formatName(name, prefix, c.App.separator, c.App.delimiters...)
}
```

Or do I miss any other usage of this flag caching behaviour?

Reply:
I can easily remove the code mentioned and simplify the struct from `map[string]*Flag` to `[]Flag` and everything works just fine.
The reason that I wanted to cahce this behaviour is because every time the `Flag(name)` function is called, which can happen in other cases than the creation of flag, it would re-register the environment variable and append it to the command's array of flags.
However, in our case, the function is only called in flag creation, which simplifies the purpose and behaviour of it. 
Therefore, I removed the caching part and simplified the map back to a slice.